src/main/java/reactor/kafka/receiver/internals/ConsumerFlux.java

```diff
@@ -65,43 +65,44 @@
         "endOffsets"
     ));
 
-    final AtomicBoolean isActive = new AtomicBoolean();
+    private final AtomicBoolean isActive = new AtomicBoolean();
 
-    final AtomicBoolean isClosed = new AtomicBoolean();
+    private final AtmostOnceOffsets atmostOnceOffsets = new AtmostOnceOffsets();
 
-    final AtomicBoolean awaitingTransaction = new AtomicBoolean();
+    private final CommitEvent commitEvent = new CommitEvent();
 
-    final AtmostOnceOffsets atmostOnceOffsets = new AtmostOnceOffsets();
+    private final PollEvent pollEvent;
 
-    final PollEvent pollEvent;
+    private final AckMode ackMode;
 
-    final AckMode ackMode;
+    private final ReceiverOptions<K, V> receiverOptions;
 
-    final ReceiverOptions<K, V> receiverOptions;
+    private final ConsumerFactory consumerFactory;
 
-    final ConsumerFactory consumerFactory;
+    private final Predicate<Throwable> isRetriableException;
```

Comment:
Oh, sorry for making it harder to follow. Actually, let me re-arrange some of the fields to make it easier (also for the future us)

`isClosed` was removed since it wasn't needed anymore, and `isRetriableException` was added, otherwise it was impossible to avoid accessing the internal state from tests (previously, they were creating a new `commitEvent` (sic!) with an overridden method

Reply:
should be much better now